worker-document/src/main/java/com/hpe/caf/worker/document/tasks/AbstractTask.java

```diff
@@ -93,4 +104,58 @@ public final WorkerResponse handleGeneralFailure(final Throwable failure)
 
     @Nonnull
     protected abstract WorkerResponse handleGeneralFailureImpl(final Throwable failure);
+
+    private static class ResponseOptionsImpl implements ResponseOptions
```

Comment:
- I don't see the advantage of this being a nested class.  Why not just create a full proper class for it alongside the other object model implementations?
- Why not extend `DocumentWorkerObjectImpl` to implement `getApplication()`?  It doesn't save you much but all the other object model implementations use it.

Reply:
Nothing outside this class creates it. This was the reasoning. I was thinking about it but then decided I prefer this type of structure. I'm open to changing it if you give me advantage of doing so.